client/cache.go

```diff
@@ -0,0 +1,117 @@
+package client
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/drand/drand/log"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// NewCachingClient is a meta client that stores an LRU cache of
+// recently fetched random values.
+func NewCachingClient(client Client, size int, log log.Logger) (Client, error) {
+	cache, err := lru.New(size)
```

Comment:
I would propose using ARC cache from this package. It solves problems with cache trashing.

Reply:
done